components/places/src/storage/history.rs

```diff
@@ -483,65 +578,24 @@ pub fn delete_everything(db: &PlacesDb) -> Result<()> {
 }
 
 fn delete_place_visit_at_time_in_tx(db: &PlacesDb, url: &str, visit_date: Timestamp) -> Result<()> {
-    let place = db.conn().try_query_row(
-        "SELECT h.id
-         FROM moz_places h
-         JOIN moz_historyvisits v
-           ON v.place_id = h.id
-         WHERE v.visit_date = :visit_date
-           AND h.url_hash = hash(:url)
-           AND h.url = :url
-         LIMIT 1",
-        &[
-            (":url", &url as &dyn rusqlite::ToSql),
-            (":visit_date", &visit_date),
-        ],
-        |row| row.get::<_, RowId>(0),
-        true,
-    )?;
-
-    let place_id = if let Some(id) = place {
-        id
-    } else {
-        // No such visit, nothing to do.
-        return Ok(());
-    };
-
-    db.conn().execute_cached(
-        "INSERT OR IGNORE INTO moz_historyvisit_tombstones(place_id, visit_date)
-         VALUES(:place_id, :visit_date)",
-        &[
-            (":place_id", &place_id as &dyn rusqlite::ToSql),
-            (":visit_date", &visit_date),
-        ],
-    )?;
-
-    db.conn().execute_cached(
-        "DELETE FROM moz_historyvisits
-         WHERE visit_date = :visit_date
-           AND place_id = :place_id",
-        &[
-            (":place_id", &place_id as &dyn rusqlite::ToSql),
-            (":visit_date", &visit_date),
-        ],
-    )?;
```

Comment:
Note that there's a subtle difference between the current code and the new code if 2 visits have the same `place_id` and `visit_date`.

The current code deletes all visits with the same `visit_date`, but the new code only deletes 1 timestamp.  This seems more correct to me, since we're only inserting a single tombstone, but I wasn't really sure how that case should be handled.

Another option that makes sense to me would be to drop the `LIMIT 1` part from the query.  That way if there's multiple visits with the same `visit_date`, we'll delete all rows and insert that many rows into the tombstones table.

It's hard to say what's correct, especially since this corner case seems almost impossible in practice.